### ISSUE TITLE
[WavLM] How can I get padding_mask and applying SpecAugment like masking

### ISSUE THREAD

**Issue by LEECHOONGHO** [microsoft/unilm#649]
Hello, I'm trying to train ASR model with WavLM's modules.
For training task, I'm trying to make get_mask_from_lengths function for padding_mask like below.
There seems to be no error and the output seems to be normal.
I'm not sure if this is the right way to make padding_mask.
```
import torch, time, random
from model.WavLM import WavLM, WavLMConfig
cfg = WavLMConfig()
model = WavLM(cfg)
model.eval()

def get_mask_from_lengths(lengths, max_len=None):
    with torch.no_grad():
        batch_size = lengths.shape[0]
        if max_len is None:
            max_len = torch.max(lengths).item()

        ids = torch.arange(0, max_len).unsqueeze(0).expand(batch_size, -1).to(lengths.device)
        mask = ids >= lengths.unsqueeze(1).expand(-1, max_len)

    return mask

# Generate Dummy Audio with random audio lengths
batch_size = 16
max_len = random.randint(1, 80000)
wav_input_16khz = torch.randn(batch_size , max_len)
wav_lens = torch.randint(1, max_len, (batch_size ,))

wav_lens = torch.div(wav_lens, 320, rounding_mode='floor')
padding_mask = get_mask_from_lengths(wav_lens, max_len // 320)

# padding_mask : torch.Tensor([batch_size, max_seq_len])

rep, _ = model.extract_features(wav_input_16khz, padding_mask=padding_mask , mask=True)
```

And If I want to apply SpecAugment for ASR training,
Can I use apply_mask function in WavLM model for applying `SpecAugment` by switching [x[mask_indices] = self.mask_emb](https://github.com/microsoft/unilm/blob/master/wavlm/WavLM.py#:~:text=x%5Bmask_indices%5D%20%3D%20self.mask_emb) to x[mask_indices] = 0? 

Any advice would be appreciated!

**Comment by MarkWuNLP** [comment]
@LEECHOONGHO  Sorry for the late reply. Here are two ways to fine-tune WavLM on ASR, including the SUPERB approach and the wav2vec approach. Which one do you need.

For your second question, No we cannot change it to x[mask_indices] = 0, since it will hurt the model performance.
